GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1612,7 +1582,7 @@ private FSResult RemoveFolderPlaceholderIfEmpty(IPlaceholderData placeholder)
             string childName;
             string parentKey;
             this.GetChildNameAndParentKey(placeholder.Path, out childName, out parentKey);
-            PathSparseState sparseState = this.GetPathSparseState(placeholder.Path);
+            PathSparseState sparseState = this.GetFolderPathSparseState(parentKey);

```

Comment:
Can this line be removed completely?  It looks like `sparseState` is only getting used for this check:

`sparseState != PathSparseState.Included`

And as best I can tell `this.TryGetSha` will return false for folders that are not included (it uses `GetProjectedFolderEntryData`).  If we can remove this call it will save us a lot of walking the tree in the more common scenario that placeholders on disk are still included.

Reply:
Looks like it.  I will give it a try and see.